### ISSUE TITLE
No browsers except Safari are available

### ISSUE THREAD

**Issue by scotartt** [will-stone/browserosaurus#121]
Browsersaurus version is v6.1.0 on macOS 10.14.6

The only browsers available to it are "Safari" and "Copy to clipboard"

I've got Google Chrome and Firefox installed. I've restarted it and such, it's always like this.

![Screenshot 2019-09-25 13 10 13](https://user-images.githubusercontent.com/2726118/65566350-db11aa80-df95-11e9-8567-e99789755a02.png)

![Screenshot 2019-09-25 13 09 10](https://user-images.githubusercontent.com/2726118/65566352-dea53180-df95-11e9-99d0-0484ea0af89f.png)

![Screenshot 2019-09-25 13 09 31](https://user-images.githubusercontent.com/2726118/65566355-e369e580-df95-11e9-825f-20cc1b811316.png)

**Comment by scotartt** [comment]
It updated itself to 6.4.0 and restarted but has the same result.

**Comment by will-stone** [comment]
Hi @scotartt. I know it's annoying but can you please try restarting your Mac? After restarting and launching B, please count to ten before either clicking a link or opening the menu bar menu. Let me know how that goes.

**Comment by scotartt** [comment]
Hi ... nope didn't work at all.

**Comment by will-stone** [comment]
This means the system profiler is unable to find your apps. Are you installing them outside the Apps folder? Basically, if this command `system_profiler -xml SPApplicationsDataType` cannot find your browsers, Browserosaurus won't be able to either.

**Comment by scotartt** [comment]
They are all installed in Applications, but that command (without the -xml formatting) produces no results:

```
$ system_profiler SPApplicationsDataType
$ ls /Applications
Browserosaurus.app  
Safari.app
Firefox.app                    
Google Chrome.app              
```

**Comment by scotartt** [comment]
does spotlight indexing need to be enabled?

**Comment by will-stone** [comment]
Ooo, maybe. I'm not sure where the profiler gets its info from.

**Comment by ralphsun73221** [comment]
I have the same problem

But I can find firefox but can't find brave

**Comment by will-stone** [comment]
Can you please try Apple's recommended way of reindexing Spotlight: https://support.apple.com/en-gb/HT201716

In this case, we need it to reindex the Applications folder (if that's where you have Brave installed).

**Comment by scotartt** [comment]
I did three things and it finally worked.

0. step 0: quit Browsersaurus (I also actually uninstalled it).

1. as per the recommendation on the apple website above, except that I dragged the entire disk into the excluded folders, then removed it.

2. In a terminal shell, `sudo mdutil -i on /`

3. I also have the 'CleanMyMac X' app, and that has the ability to also force a reindex of spotlight, as documented here: https://macpaw.com/how-to/reindex-spotlight (indeed, I got the terminal commands from that webpage).

Anyway ... that sequence of actions finally fixed it after I restarted Browsersaurus. I was, as a last-ditch resort, also going to try to use `launchctl` to turn spotlight off then on, but didn't need to. As an aside, I also ran a bunch of other optimisations that CleanMyMac has, I guess there's some low probability that that one of those was the actual cure, but, it's probably spotlight (seems like most likely).

Also, now `system_profiler SPApplicationsDataType` outputs a bunch of stuff.

Also in this area, I use LaunchBar, which was why I hadn't noticed that the Spotlight index was totally hosed.
